ui/src/components/category/Category.tsx

```diff
@@ -50,7 +53,7 @@ export default function Category({ userEmail }: CategoryProps) {
                         </TableRow>
                     </TableHead>
                     <TableBody>
-                        {!loading && data && data.map((category: Category) => (
+                        {!loading && Array.isArray(data) && data.map((category: Category) => (
```

Comment:
This should ne be needed as we except data to be present and to be either empty list or have objects, if categories return not empty list, need to fix it.